qrexec/policy/parser.py

```diff
@@ -1651,10 +1693,19 @@ def collect_targets_for_ask(self, request):
                 rule_target = (
                     getattr(rule.action, "target", None) or rule.target
                 )
-                expansion = set(
-                    rule_target.expand(system_info=request.system_info)
-                )
-
+                expansion = set()
+                for potential_target in rule_target.expand(
+                        system_info=request.system_info):
+                    try:
+                        # The policy agent cannot handle UUIDs (and rightly so,
+                        # those are meaningless to humans). Convert them to names.
+                        # VM names can be reused, but in practice, this is unlikely
+                        # to happen except by user request or DispVM name reuse.
+                        # The latter will not happen for a week and so is unlikely
+                        # to confuse humans.
```

Comment:
I think you ignored first sentence of my earlier comment. Even if VM name gets reused, you translate policy agent response back to UUID using dict collected at the request time. So, if VM with that  **UUID** doesn't exist anymore, the call will fail, it won't make it to the same-named new VM. DispVMs are not any different here.
Have I missed anything?

BTW, it might be useful to add yet another dict to the agent params, with name->uuid mapping of the possible targets (+ source too?), just in case some future/alternative agent would like to use it. But due to above, other parts can still use names, so it won't break compatibility with old agent.